lib/widgets.php

```diff
@@ -195,6 +196,7 @@ function gutenberg_get_legacy_widget_settings() {
 				'name'              => html_entity_decode( $widget_obj['name'] ),
 				'description'       => html_entity_decode( wp_widget_description( $widget_id ) ),
 				'isReferenceWidget' => true,
+				'blockName'         => 'core/legacy-widget-' . preg_replace( '#[^a-zA-Z0-9-]#', '-', $widget_id ),
```

Comment:
I don't think we've ever done this but I wonder if we should use sub-namespaces here? In other words, the name would be `core/legacy-widget/marquee`.